Remove ability to include commands directly on a message

lukemelia opened the pull request:
- Introduce waitForMatrixEvent, waitForCompletedCommandRequest utils
- Add GetEventsFromRoom command that is used to implement waitForMatrixEvent util
- Removes ability to include commands directly on a message
- Removed Command#waitForNextCompletion
- Removed "invocations" internals of Command that are no longer needed

lukemelia committed:
Remove ability to include commands directly on a message

IanCal reviewed (approved):
Looks good, I assume there are some things in the ai bot that can be removed too?

I think it'd be fine to just delete the things we're mostly commenting out in the experiments realm, we're changing things enough that removing things we don't support any more should be alright. Product requirements and app gen we don't have any more, really.

lukemelia commented:
> Looks good, I assume there are some things in the ai bot that can be removed too?

I don't think things look any different to the AI Bot. Am I forgetting something?

lukemelia commented:
> I don't think things look any different to the AI Bot. Am I forgetting something?

I see that are close to removing context.tools on CardMessageContent. We need to eliminate the auto per-type patch commands before we do though.

lukemelia committed:
Introduce waitForMatrixEvent, waitForCompletedCommandRequest utils

- Add GetEventsFromRoom command that is used to implement waitForMatrixEvent util
- Removed Command#waitForNextCompletion
- Removed "invocations" internals of Command that are no longer needed

lukemelia commented:
Please check the changes since your last review.

lukemelia committed:
Fix extraneous use of commands

lukemelia committed:
Add waitForRealmState util to command utils
